plugin.audio.cbcradio/default.py

```diff
@@ -0,0 +1,226 @@
+    play_item.setInfo('music', info_labels)
+    return play_item
+
+
+def set_program_art(program):
+    play_item = PLAYER.getPlayingItem()
+    if program is not None:
+        try:
+            play_item.setArt({'fanart': program.artwork_url})
+            xbmc.log(level=xbmc.LOGDEBUG, msg=f"plugin.audio.cbcradio: Set 'fanart': {program.artwork_url}")
+        except Exception:
+            xbmc.log(level=xbmc.LOGDEBUG, msg=f"plugin.audio.cbcradio: Could not fetch fanart: {program.artwork_url} Using default.")
+            play_item.setArt({'fanart': FANART})
+    else:
+        play_item.setArt({'fanart': FANART})
+        xbmc.log(level=xbmc.LOGDEBUG, msg=f"plugin.audio.cbcradio: Set 'fanart': FANART")
+    PLAYER.updateInfoTag(play_item)
+
+
+def news_break():
+    xbmc.log(level=xbmc.LOGDEBUG, msg="plugin.audio.cbcradio: News break.")
+    play_item = PLAYER.getPlayingItem()
+    play_item.setArt({'fanart': FANART})
+    xbmc.log(level=xbmc.LOGDEBUG, msg="plugin.audio.cbcradio: Set 'fanart': FANART")
+    tag = play_item.getMusicInfoTag()
+    tag.setAlbum(None)
+    tag.setArtist(None)
+    tag.setTitle("CBC News")
+    tag.setComment(None)
+    PLAYER.updateInfoTag(play_item)
+
+
+def calc_minutes():
+    time_now = datetime.now().strftime("%M")
+    mins = int(time_now)
+    return mins
+
+
+def initialize(key, location, url):
+    program_schedule = now_playing.get_program_schedule(key, location)
+    program = now_playing.get_current_program(program_schedule)
+    playlog = now_playing.get_playlog(program, location)
+    last_track = now_playing.get_current_track(playlog)
+    play_item = create_play_item(last_track)
+    play_item.setPath(url)
+    play_item.setProperty('IsPlayable', 'true')
+    play_item.addStreamInfo('audio', {'codec': 'aac', 'channels': 2})
+    if PLAYER.isPlaying():
+        PLAYER.stop()
+        xbmc.sleep(1000)
+    xbmcplugin.setResolvedUrl(ADDON_HANDLE, True, listitem=play_item)
+    while not PLAYER.isPlaying():
+        xbmc.sleep(1000)
+    while xbmc.getCondVisibility('Window.IsActive(BusyDialog)'):
+        xbmc.sleep(1000)
+        xbmc.log(level=xbmc.LOGDEBUG, msg="plugin.audio.cbcradio: sleep for fullscreen")
+    set_program_art(program)
+    xbmc.executebuiltin('Action(FullScreen)')
+    xbmc.log(level=xbmc.LOGDEBUG, msg="plugin.audio.cbcradio: fullscreen")
+    return program_schedule, program, playlog, last_track, play_item
+
+
+def play_stream(key, location, url):
+    program_schedule, program, playlog, last_track, play_item = initialize(key, location, url)
+    c = -1
+    while not MONITOR.abortRequested():
+        now = time.time() * 1000
+        if calc_minutes() in [00, 1, 2, 3, 4, 5] and key == 2:
+        # cbc music has 5 min news breaks at the top of the hour
+            news_break()
+            while calc_minutes() in [00, 1, 2, 3, 4, 5]:
+                xbmc.sleep(10000)
```

Comment:
This call completely block execution for 10 seconds without the option to abort. You should also use the `MONITOR.waitForAbort()` here and test for abort afterwards. Just like in line 198

Reply:
done